lib/SILGen/SILGenApply.cpp

```diff
@@ -2179,13 +2179,14 @@ buildBuiltinLiteralArgs(SILGenFunction &SGF, SGFContext C,
 
   case MagicIdentifierLiteralExpr::Line:
   case MagicIdentifierLiteralExpr::Column: {
-    auto Loc = getLocInOutermostSourceFile(SGF.getSourceManager(),
-                                           magicLiteral->getStartLoc());
     unsigned Value = 0;
-    if (Loc.isValid()) {
-      Value = magicLiteral->getKind() == MagicIdentifierLiteralExpr::Line
-                  ? ctx.SourceMgr.getPresumedLineAndColumnForLoc(Loc).first
-                  : ctx.SourceMgr.getPresumedLineAndColumnForLoc(Loc).second;
+    if (auto Loc = magicLiteral->getStartLoc()) {
+      Loc = getLocInOutermostSourceFile(SGF.getSourceManager(), Loc);
+      if (Loc.isValid()) {
```

Comment:
This looks like a reasonable fix for the issue that `magicLiteral->getStartLoc()` may sometimes be invalid and passing an invalid `SourceLoc` to `getLocInOutermostSourceFile()` will trip an assertion.